### PR TITLE
Support setting ephemeral storage

### PR DESCRIPTION
**Pull request by katrogan**
# TL;DR
Now that k8s supports setting ephemeral storage we should expose that as a configurable setting to end users. At the very least this change allows doing so in pod tasks.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/494

## Follow-up issue
_NA_

### CODE REVIEW

**Comment by katrogan** [comment]
cc @jeevb

**Comment by jeevb** [comment]
Awesome! 🙏

I imagine this will require changes in flytekit (and flyteidl)?

**Review by EngHabu — COMMENTED** [review]
cool! a couple of questions:
* A quick google search shows me this [result](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#local-ephemeral-storage) which suggests it's still in beta? but then it says 1.10... so I'm confused... mind double checking...
* If it only went to stable recently, should we have a config here that enables this behavior? otherwise it goes back to the old behavior... and maybe the default value is the new behavior?

**Comment by katrogan** [comment]
@jeevb yup, working on those changes in parallel

@EngHabu hah yeah we went through the same confusion, see https://flyte-org.slack.com/archives/CREL4QVAQ/p1624469230251200?thread_ts=1624468996.250400&cid=CREL4QVAQ

it should now be supported on 1.19+

**Review by EngHabu — APPROVED** [review]
